src/ThriveWorkerUnit.sol

```diff
@@ -0,0 +1,134 @@
+        address _assignedAddress,
+        string memory _metadata
+    ) {
+        require(_moderator != address(0), "Moderator address is required");
+        require(_deadline > block.timestamp, "Deadline must be in the future");
+
+        moderator = _moderator;
+        rewardToken = _rewardToken;
+        rewardAmount = _rewardAmount;
+        maxRewards = _maxRewards;
+        deadline = _deadline;
+        maxCompletionsPerUser = _maxCompletionsPerUser;
+        assignedAddress = _assignedAddress;
+        metadata = _metadata;
+
+        status = Status.Active;
+    }
+
+    /**
+     * @notice Updates the status of the work unit to Expired if the deadline has passed.
+     */
+    function updateStatus() external {
+        if (block.timestamp > deadline) {
+            status = Status.Expired;
+        }
+    }
+
+    /**
+     * @notice Deposit native tokens into the contract.
+     */
+    function deposit() external payable {
+        require(msg.value > 0, "Deposit amount must be greater than zero");
+    }
+
+    /**
+     * @notice Allows depositing native tokens without data.
+     */
+    receive() external payable {}
+
+    /**
+     * @notice Allows depositing native tokens with arbitrary data.
+     */
+    fallback() external payable {}
+
+    /**
+     * @notice Rewards a user for completing the work unit.
+     * @param _recipient Address of the contributor.
+     * @param _reason Reason for the reward.
+     */
+    function reward(address _recipient, string calldata _reason) external {
+        require(block.timestamp <= deadline, "Work unit has expired");
+        require(
+            completions[_recipient] < maxCompletionsPerUser,
+            "Max completions per user reached"
+        );
+        require(
+            balanceOf[_recipient] + rewardAmount <= maxRewards,
+            "Reward pool exceeded"
+        );
+        require(address(this).balance >= rewardAmount, "Insufficient contract balance");
+
+        if (assignedAddress != address(0)) {
+            require(
+                _recipient == assignedAddress,
+                "Work unit restricted to a specific address"
+            );
+        }
+
+        completions[_recipient] += 1;
+        balanceOf[_recipient] += rewardAmount;
+
+        emit Reward(_recipient, rewardAmount, _reason);
+    }
+
+    /**
+     * @notice Allows a user to withdraw their earned rewards.
+     * @param _amount Amount to withdraw.
+     */
+    function withdraw(uint256 _amount) external {
+        require(balanceOf[msg.sender] >= _amount, "Insufficient balance");
+
+        balanceOf[msg.sender] -= _amount;
+
+        if (rewardToken == address(0)) {
+            require(address(this).balance >= _amount, "Insufficient contract balance");
+            payable(msg.sender).transfer(_amount);
+        } else {
+            IERC20(rewardToken).transfer(msg.sender, _amount);
+        }
+
+        emit Withdrawal(msg.sender, _amount);
+    }
```

Comment:
these 2 aren't needed, we need function to create confirmations, e.g. something like this:
```solidity
function _confirm (address contributor, string memory metadata) internal virtual {
 // check max completions
 // send rewards to both contributor and validator
 // emit event
 // store contribution in structure

}

function confirm (address contributor, string memory metadata) onlyValidator external {
 _confirm(contributor, metadata);
}
```

Reply:
okay, added badge eligibility also imagining that we will deploy a contract that will implement IBadgeQuery interface and we can check using that. If you think thats not needed for now I can delete that.